client/verta/verta/_repository/commit.py

```diff
@@ -548,10 +548,32 @@ def revert(self, other=None, message=None):
         elif self._repo.id != other._repo.id:
             raise ValueError("Commit and `other` must belong to the same Repository")
 
-        if message is None:
-            message = "Revert {}".format(other.id[:7])
+        msg = _VersioningService.RevertRepositoryCommitsRequest()
+        msg.base_commit_sha = self.id
+        msg.commit_to_revert_sha = other.id
+        if message is not None:
+            msg.content.message = message
 
-        self.apply_diff(other.get_revert_diff(), message)
+        data = _utils.proto_to_json(msg)
+        endpoint = "{}://{}/api/v1/modeldb/versioning/repositories/{}/commits/{}/revert".format(
+            self._conn.scheme,
+            self._conn.socket,
+            self._repo.id,
+            msg.commit_to_revert_sha,
```

Comment:
This will always revert the commit onto itself, which is different from what we had before.

Reply:
Not entirely sure what you mean by "revert the commit onto itself" (is the revert commit a child of the correct commit?),
but I'm following the protos: https://github.com/VertaAI/modeldb/blob/master/protos/protos/public/modeldb/versioning/VersioningService.proto#L630

Reply:
The protos have option to use two different commits. You are only using one, from what I can see.

Reply:
Oh, I can't? Then how exactly is this supposed to be used?

cc @ravishetye if he also has anything to add

Reply:
the base_commit commit on which the revert is to be applied is coming from the body

Reply:
https://github.com/VertaAI/modeldb/blob/master/protos/protos/public/modeldb/versioning/VersioningService.proto#L416

Reply:
>You are only using one
Is this accounting for lines 552 and 553, where I'm using two commit IDs?

>You can't revert `C1` and apply the changes to `C2`.
Do you mean "you can't" as in "it's impossible", or as in "you're doing this incorrectly"? Because based on the [backend PR](https://github.com/VertaAI/modeldb/pull/497#pullrequestreview-391257314), I think this should be possible.

Reply:
Oh, I missed that you are setting the two fields, as that's unexpected to me =) I thought one would be set only by the path and completely missed that you set another in the other field. Ignore my comment.